### ISSUE TITLE
elasticsearch script does not work with arguments that have spaces

### ISSUE THREAD

**Issue by jaymode** [elastic/elasticsearch#12677]
If you attempt to start elasticsearch and have an argument that has a space in it, the argument is not parsed correctly and the value after the space is treated as a separate argument.

```
$ bin/elasticsearch -Des.pidfile="/path/with space/es.pid"
ERROR: Parameter [space/es.pid]does not start with --
```

The fix is probably similar to the ones in #12508

**Comment by xuzha** [comment]
I'm not quite sure we should change this, user could escape the ' " ' by doing this 
```
bin/elasticsearch -Des.pidfile=\"/path/with space/es.pid\"
``

**Comment by jaymode** [comment]
Thanks for opening a PR @xuzha. I think having to escape the quotes on the command line makes it less user friendly. When I think of running commands with spaces, I would expect to need to escape spaces or put quotes around the item with spaces; I didn't expect that I'd need to escape the quotes.

**Comment by clintongormley** [comment]
Closed by #12710
